src/master/chunks.cc

```diff
@@ -378,7 +380,7 @@ class Chunk {
 			all.addPart(part.type, csdb_find(part.csid)->label);
 		}
 
-		all.optimize();
+		all.optimize(gUseLinearAssignmentOptimizer, &gLinearAssignmentCache);
```

Comment:
Please consider that both arguments are global variables. Probably there is no need to pass them as arguments or they can be moved to a place where the optimize function can find them.